python_modules/libraries/dagster-dg/dagster_dg/context.py

```diff
@@ -800,3 +812,89 @@ def _validate_plugin_entry_point(context: DgContext) -> None:
             """,
             context.config.cli.suppress_warnings,
         )
+
+
+DG_UPDATE_CHECK_INTERVAL = datetime.timedelta(hours=1)
+DG_UPDATE_CHECK_ENABLED_ENV_VAR = "DAGSTER_DG_UPDATE_CHECK_ENABLED"
+
+
+@whitelist_for_serdes
+@record
+class DgPyPiVersionInfo:
+    timestamp: float
+    raw_versions: list[str]
+
+    @property
+    def datetime(self) -> datetime.datetime:
+        return datetime.datetime.fromtimestamp(self.timestamp)
+
+    @cached_property
+    def versions(self) -> list[Version]:
+        return sorted(Version(v) for v in self.raw_versions)
+
+
+def _validate_dg_up_to_date(context: DgContext) -> None:
+    # Don't check if we've disabled the check
+    if not (
+        bool(int(os.getenv(DG_UPDATE_CHECK_ENABLED_ENV_VAR, "1")))
+        and "dg_outdated" not in context.config.cli.suppress_warnings
+    ):
+        return
+
+    version_info = _get_dg_pypi_version_info(context)
+    if version_info is None:  # Nothing cached and network error occurred
+        return None
+
+    installed_version = Version(__version__)
+    latest_version = version_info.versions[-1]
+    if installed_version < latest_version:
+        emit_warning(
+            "dg_outdated",
+            f"""
+            There is a new version of `dagster-dg` available:
+
+                Latest version: {latest_version}
+                Installed version: {installed_version}
+
+            Update your dagster-dg installation to keep up to date:
+
+                [uv tool]  $ uv tool upgrade dagster-dg
```

Comment:
Definitely not necessary to get this in, but can we detect which of these a user is using contextually? I might not know offhand which I'm using especially if I'm not the one who set up the dg project